test/lib/ansible_test/_data/completion/remote.txt

```diff
@@ -2,7 +2,7 @@ alpine/3.20 python=3.12 become=doas_sudo provider=aws arch=x86_64
 alpine become=doas_sudo provider=aws arch=x86_64
 fedora/40 python=3.12 become=sudo provider=aws arch=x86_64
 fedora become=sudo provider=aws arch=x86_64
-freebsd/13.3 python=3.9,3.11 python_dir=/usr/local/bin become=su_sudo provider=aws arch=x86_64
+freebsd/13.4 python=3.11 python_dir=/usr/local/bin become=su_sudo provider=aws arch=x86_64
```

Comment:
Would it be possible to split up removal and adding the new version into two different PRs, to make life for collection maintainers a bit less hard? (Same for the other similar issues/PRs.)

Reply:
@felixfontein We want to avoid a delay between the addition and removal. Is there some benefit if two PRs are used, with one merged immediately after the other?

Reply:
There's a benefit if they are NOT merged directly after each other, say a week apart. Then collections have a week to modify their CI to switch to the new version.

Reply:
This one's tricky- if it were a completely mechanical process that could be easily automated, I'd suggest a compromise like "add new platforms to devel as they come up, and rip out all the old ones in a single commit when we create the stable branch". Unfortunately, that hasn't been our experience- nearly every platform support change requires some kind of bespoke customization/refactoring of the platform-specific bootstrap stuff. We've been trying to cross-train other members of the core team to distribute the load for this process to people other than @mattclay, and the two-pass async workflow part of it, as well as the small window where we have to actively support a new platform and one that's going away, were significant stumbling blocks. This is also part of the reason the `milestone` ref exists- we've had similar requests from other teams/projects where they wanted to gate all core devel PRs on their own CI passing, which would be yet another significant impediment to our velocity. At the end of the day, drinking from the `devel` firehose *will* sometimes break things, and we're under-resourced enough that `milestone` is the only point of deferred entropy that we can commit to right now.

Reply:
> nearly every platform support change requires some kind of bespoke customization/refactoring of the platform-specific bootstrap stuff

The customizations I've seen so far didn't become easier if support for the previous version is removed at the same time.

> This is also part of the reason the milestone ref exists-

The thing is: `milestone` does not improve this situation. Same as `devel`, at a point in time the old CI config suddenly stops working and CI is broken until the new platform is added to CI, which can take some time for pretty much the same reason that adding a new platform to ansible-core takes some time. The main difference is that you know in advance when that will happen.

In fact `milestone` is way worse than `devel`: Instead of having to deal with 1-2 platforms at the same time, you have to deal with quite a few platforms at the same time, which increases the amount of work needed to get CI working again.

(The only way to avoid this is to use neither `devel` nor `milestone`, and do all the work once a new `stable-2.x` branch is there. It will still be a lot of work, but at least you can do that work without impacting the current CI - the new stable branch is only added to everyone's CI once you merge the PR. This obviously has the disadvantage that you only find out about all changes in `devel` by the time everything's already very final...)